### PR TITLE
Show total ready and unacked messages

### PR DESCRIPTION
**Pull request by Christina-84**
Adds total number of ready and unacked messages for all queues to **stats** on **nodes page**

image

### CODE REVIEW

**Comment by jage** [comment]
"Docker images / main (pull_request)" is failing due to credentials being ignored since you were not a collaborator on this repo when the PR was opened. I've added you and tried to rerun but still fails, so let's just ignore it here as it will (most likely) work next time.

File: static/js/nodes.js [hunk 1]
```diff
@@ -9,27 +9,38 @@
...
-  if (data === null) {
-    update(render)
-  }
 
   function update (cb) {
-    lavinmq.http.request('GET', url).then(function (response) {
-      data = response
-      render(response)
+    const statsPromise = lavinmq.http.request('GET', url)
+    const queueStatsPromise = lavinmq.http.request('GET', '/api/queues')
```
Comment (jage) [inline static/js/nodes.js:15]: We might want to do this sum server side, if we have a lot of queues. Maybe we can do an experiment with 10k queues how long this takes here? :)

Comment (Christina-84) [reply static/js/nodes.js:15]: Yes true! Ill try putting it server side and re-iterate 👍

File: static/js/nodes.js [hunk 2]
```diff
@@ -9,27 +9,30 @@
...
-  if (data === null) {
-    update(render)
-  }
 
   function update (cb) {
-    lavinmq.http.request('GET', url).then(function (response) {
-      data = response
-      render(response)
+    const statsPromise = lavinmq.http.request('GET', url)
+    Promise.all([
```
Comment (carlhoerberg) [inline static/js/nodes.js:15]: `Promise.all` for a single item is necessary.

File: static/js/nodes.js [hunk 3]
```diff
@@ -9,27 +9,30 @@
...
+      statsPromise
+    ]).then(function (responses) {
```
Comment (carlhoerberg) [inline static/js/nodes.js:17]: use the `(response) => {}` syntax.

File: static/js/nodes.js [hunk 4]
```diff
@@ -9,27 +9,30 @@
...
+      const statsData = responses[0]
```
Comment (carlhoerberg) [inline static/js/nodes.js:18]: are you sure there will always be one item? and why is an array returned if there's only a single item?

File: static/js/nodes.js [hunk 5]
```diff
@@ -9,27 +9,30 @@
...
   }
 
-  function render (data) {
+  function render (data, queueData) {
     document.querySelector('#version').innerText = data[0].applications[0].version
+
     for (const node of data) {
       updateDetails(node)
-      updateStats(node)
+      if (queueData instanceof Object) {
```
Comment (carlhoerberg) [inline static/js/nodes.js:31]: when is it not?

**Review by carlhoerberg — CHANGES_REQUESTED** [review]

**Comment by Christina-84** [comment]
Was eager and pushed half ready, now i think the node.js file makes more sense

File: static/js/nodes.js [hunk 6]
```diff
@@ -27,6 +21,7 @@
 
   function render (data) {
     document.querySelector('#version').innerText = data[0].applications[0].version
+
```
Comment (carlhoerberg) [inline static/js/nodes.js:24]:
```suggestion
```

**Review by carlhoerberg — APPROVED** [review]

**Review by jage — APPROVED** [review]
Great!

**Comment by jage** [comment]
Feel free to squash merge, and add a row under unreleased to `Changelog`, either in this branch or directly in `main`.

**Pull request merged**